src/yaml-tools.js

```diff
@@ -111,11 +123,30 @@ yamlLibrary = {
   // doc: The SAM.YML parsed document
   // schema: the JSON schema to compare the doc to
   checkDocAgainstSchema: function (doc, schema, failed, warning, info) {
-    this.doc = doc;
+    this.doc = this.docs[doc];
     var keys = [];
 
     this.recurseIntoArray(schema, keys, null, failed, warning, info);
+  },
+
+  loadDocData: function (docs) {
+    let docsArray = cleanDocsData(docs).split('---');
+
+    for (let i = 0; i < docsArray.length; i++) {
+      let allComments = true;
+      let yamlDoc = yaml.load(docsArray[i]);
+      this.docs[i] = yamlDoc == null || yamlDoc == undefined ? {} : yamlDoc;
+
+      let lines = docsArray[i].split('\n');
+      for (let j = 0; j < lines.length; j++) {
+        let line = lines[j].trim();
```

Comment:
Could you do this trim as part of the cleanDocsData?

Reply:
Actually, I can't do a JS `trim` for anything but comment checking, because of YAML white space.  But I don't need it anymore, because of the cleaning method.